lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2817,25 +2841,20 @@ public function createEntity($className, array $data, &$hints = [])
             }
 
             $this->originalEntityData[$oid] = $data;
+
+            if ($entity instanceof NotifyPropertyChanged) {
+                $entity->addPropertyChangedListener($this);
+            }
```

Comment:
How does this change and the ones below relate to the safeguard part?

Reply:
So… does it relate to the safeguard part or not? I'm confused now.

Reply:
Yes, it belongs here. It makes sure we run the safeguard check also for entities created in this method.